app/src/main/java/org/simple/clinic/home/patients/PatientScreenRepository.kt

```diff
@@ -2,22 +2,46 @@ package org.simple.clinic.home.patients
 
 import io.reactivex.Observable
 import io.reactivex.rxkotlin.ofType
+import org.simple.clinic.home.patients.illustration.DayOfMonth
 import org.simple.clinic.home.patients.illustration.HomescreenIllustration
 import org.simple.clinic.storage.files.FileStorage
 import org.simple.clinic.storage.files.GetFileResult
+import org.simple.clinic.util.Just
+import org.simple.clinic.util.None
+import org.simple.clinic.util.Optional
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.util.toOptional
+import org.simple.clinic.util.toUtcInstant
+import org.threeten.bp.Instant
+import org.threeten.bp.LocalDate
 import javax.inject.Inject
 
 class PatientScreenRepository @Inject constructor(
     private val illustrationDao: HomescreenIllustration.RoomDao,
-    private val fileStorage: FileStorage
+    private val fileStorage: FileStorage,
+    private val userClock: UserClock
 ) {
 
   fun illustrations(): Observable<GetFileResult.Success> =
       illustrationDao.illustrations()
           .map { pickIllustration(it) }
+          .ofType<Just<HomescreenIllustration>>()
+          .map { it.value }
           .map { fileStorage.getFile(it.eventId) }
           .ofType()
 
-  private fun pickIllustration(illustrations: List<HomescreenIllustration>): HomescreenIllustration =
-      illustrations.first()
+  private fun pickIllustration(illustrations: List<HomescreenIllustration>): Optional<HomescreenIllustration> {
```

Comment:
I have a couple of suggestions to make to improve these two functions:

- Using an `Instant` is unnecessary since all we need are the dates. We can use `LocalDate` to compare instead.
- We can use the `.firstOrNull {}` extension on `Collection` to have a more readable method to find the right `HomescreenIllustration.

```kotlin
private fun pickIllustration(illustrations: List<HomescreenIllustration>): Optional<HomescreenIllustration> {
    val today = LocalDate.now(userClock)

    return illustrations
        .firstOrNull { illustration ->
          val showIllustrationFrom = illustration.from.toLocalDate(userClock)
          val showIllustrationTill = illustration.to.toLocalDate(userClock)

          today in showIllustrationFrom..showIllustrationTill
        }
        .toOptional()
  }

  private fun DayOfMonth.toLocalDate(userClock: UserClock): LocalDate {
    return LocalDate.now(userClock)
        .withMonth(this.month.value)
        .withDayOfMonth(this.day)
  }
```